### PR TITLE
Fix negative `take` values above the list's `maxResults` not causing an early error

### PR DESCRIPTION
**Pull request by emmatown**

### CODE REVIEW

File: packages/keystone/src/lib/core/queries/resolvers.ts [hunk 1]
```diff
@@ -200,7 +200,7 @@ function applyMaxResults(results: unknown[], list: InitialisedList, context: Key
     throw limitsExceededError({ list: list.listKey, type: 'maxResults', limit: list.maxResults });
   }
   if (context) {
-    context.totalResults += Array.isArray(results) ? results.length : 1;
+    context.totalResults += results.length;
```
Comment (mitchellhamilton) [inline packages/keystone/src/lib/core/queries/resolvers.ts:203]: (this is unrelated to the `take` thing, i just saw that this was always an array now)

**Review by timleslie — APPROVED** [review]
👍
